File: read_pdf.js

import fs from 'fs';
import pdf from 'pdf-parse/lib/pdf-parse.js';

let dataBuffer = fs.readFileSync('c:/laragon/www/Legalis-Pro-Nestor-main/public/assets/FORTALEZAS NUEVO.pdf');

try {
    const data = await pdf(dataBuffer);
    console.log("--- TEXT CONTENT ---");
    console.log(data.text);
    console.log("--- END CONTENT ---");
} catch (err) {
    console.error(err);
}

File: read_pdf.py

import re

def extract_strings(file_path):
    with open(file_path, "rb") as f:
        content = f.read()
    # Find sequences of printable characters
    strings = re.findall(rb"[\x20-\x7E]{4,}", content)
    for s in strings:
        try:
            print(s.decode("ascii"))
        except:
            pass

extract_strings(r"c:\laragon\www\Legalis-Pro-Nestor-main\public\assets\FORTALEZAS NUEVO.pdf")
